pkg/eval/options.go

```diff
@@ -37,6 +37,16 @@ func scanOptions(rawOpts RawOptions, ptr any) error {
 		if !ok {
 			return UnknownOption{k}
 		}
+
+		// An option with no value (e.g., `&a-opt`) has `$true` as its default value. However, if
+		// the option struct member is a string we want an empty string as the default value.
+		switch b := v.(type) {
+		case bool:
+			if b && structValue.Field(fieldIdx).Type().Name() == "string" {
+				v = ""
+			}
+		}
+
```

Comment:
Do not change the semantics of `&opt`. This introduces a disparity with Elvish functions, whose options do not have an intrinsic type.

Reply:
Options do have an intrinsic type but I grok your point. I added that code in the early stages of implementing this change because I was surprised by this behavior:

```
> > benchmark &min-time { }
Exception: wrong type: need string, got bool
[tty 1], line 1: benchmark &min-time { }
```

It seemed more useful to me that use case should behave as if an empty string was assigned to the string option; i.e., as if the user had typed `benchmark &min-time='' { }`. I now agree with you that an option without a value in that case should be an error if the option is not a boolean. Updated PR in a few minutes.

Reply:
And, unsurprisingly, removing that code does not affect any of the unit tests I wrote since none of them use a string option without providing a value. So removing that ill-advised change is the right thing to do. Forced push update with that change in a few minutes.

Reply:
FWIW, my original change that introduced a `time &benchmark` option relied on the behavior of this particular aspect of the PR so that someone could run `time &benchmark { code }` to enable benchmark behavior without explicitly specifying the benchmark options (thus get the benchmark defaults). It still seems useful to me, as a general principal, to allow string and numeric options to be used without a value; thus defaulting to the empty string or zero respectively. But that discussion is no longer pertinent to this change so I'm happy to punt it down the road when there is a stronger case for introducing such behavior.

Reply:
> Options do have an intrinsic type but I grok your point.

Values have intrinsic types.

In a function like `{|&opt=foo| }`, the `&opt` has no intrinsic type.